### PR TITLE
Merge: Tx Details with Tax Categories

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

We need to merge master into feature-tx-details-export to fix merge conflicts
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

**Comment by HashEngineering** [comment]
I will try to fix these merge conflicts

File: wallet/src/de/schildbach/wallet/ui/main/WalletFragment.kt [hunk 1]
```diff
@@ -93,6 +99,22 @@ class WalletFragment : Fragment(R.layout.home_content) {
...
+                val dialogFragment: TaxCategoryExplainerDialogFragment =
+                    TaxCategoryExplainerDialogFragment.newInstance(mostRecentTransaction.txId)
+                dialogFragment.show(activity?.supportFragmentManager!!, "taxcategorydialog") {
+                    val transactionDetailsDialogFragment: TransactionDetailsDialogFragment =
+                        TransactionDetailsDialogFragment.newInstance(mostRecentTransaction.txId)
+                    transactionDetailsDialogFragment.show(activity?.supportFragmentManager!!, null)
+                }
+                configuration.setHasDisplayedTaxCategoryExplainer()
+            }
+        }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/main/WalletFragment.kt:117]: before the last merge, this code was in the WalletActivity class. 

In this class, we don't have direct access to LockScreenActivity.lockScreenDisplayed.  Our goal is not to display a dialog if the wallet is logged out.  

I don't like this way of fixing this problem.

File: wallet/src/de/schildbach/wallet/ui/transactions/TransactionDetailsDialogFragment.kt [hunk 2]
```diff
@@ -20,45 +20,38 @@ import android.os.Bundle
...
     private lateinit var contentBinding: TransactionResultContentBinding
+    private val viewModel: TransactionResultViewModel by viewModels()
 
     override val backgroundStyle = R.style.PrimaryBackground
     override val forceExpand = true
 
-    @Inject
-    lateinit var walletData: WalletDataProvider
-    @Inject
-    lateinit var configuration: Configuration
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/transactions/TransactionDetailsDialogFragment.kt:59]: These are moved into the ViewModel

File: wallet/src/de/schildbach/wallet/ui/transactions/TransactionDetailsDialogFragment.kt [hunk 3]
```diff
@@ -20,45 +20,39 @@ import android.os.Bundle
...
  * @author Samuel Barbosa
  */
+@FlowPreview
 @AndroidEntryPoint
 @ExperimentalCoroutinesApi
 class TransactionDetailsDialogFragment : OffsetDialogFragment() {
 
     private val log = LoggerFactory.getLogger(javaClass.simpleName)
     private val txId by lazy { arguments?.get(TX_ID) as Sha256Hash }
-    private var tx: Transaction? = null
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/transactions/TransactionDetailsDialogFragment.kt:49]: moved to the ViewModel

File: common/src/main/res/values/styles.xml [hunk 4]
```diff
@@ -287,8 +291,8 @@
     </style>
 
     <style name="EncircledIconTheme">
-        <item name="backgroundColor">@color/steel_gray_40</item>
-        <item name="strokeColor">@color/steel_gray_40</item>
+        <item name="backgroundColor">@color/ultra_light_gray</item>
+        <item name="strokeColor">@color/ultra_light_gray</item>
```
Comment (HashEngineering) [inline common/src/main/res/values/styles.xml:295]: https://github.com/dashevo/dash-wallet/blame/4518027cefc965dafd271ede4bf42f12e2d8b1cb/common/src/main/res/values/styles.xml#L294-L295

This change was made a month ago, probably as part of a PR on this branch.

File: common/src/main/res/values/colors.xml [hunk 5]
```diff
@@ -97,6 +97,8 @@
     <color name="tangerine">#FA9269</color>
 
     <color name="dash_black_0.7">#B30C0C0D</color>
+    <color name="system_green_0.1">#1A3CB878</color>
```
Comment (HashEngineering) [inline common/src/main/res/values/colors.xml:100]: This color was also added in the CrowdNode APY PR.  We can fix those merge conflicts later!

File: common/src/main/res/values/styles.xml (hunk 4 above)
Comment (Syn-McJ) [reply common/src/main/res/values/styles.xml:295]: Yeah `steel_gray_` and `gray_` colors are obsolete

File: wallet/src/de/schildbach/wallet/ui/main/WalletFragment.kt (hunk 1 above)
Comment (Syn-McJ) [reply wallet/src/de/schildbach/wallet/ui/main/WalletFragment.kt:117]: Yeah, sometimes fragments need to listen to or check the lock screen state too.
I'm thinking we should have a base class for the fragments for this case, something like `LockScreenAwareFragment` that will get notified by the `LockScreenActivity` the same way as it notifies (or rather dismisses) all its `DialogFragments`.

**Review by Syn-McJ — APPROVED** [review]
Looks good

File: common/src/main/res/values/styles.xml (hunk 4 above)
Comment (HashEngineering) [reply common/src/main/res/values/styles.xml:295]: ok, great!

**Pull request merged**